File: backend/src/main/java/com/example/mouvement/Mouvement.java
package com.example.mouvement;

import com.example.Article;
import io.quarkus.hibernate.orm.panache.PanacheEntity;
import jakarta.persistence.*;


import java.time.LocalDateTime;

@Entity
public class Mouvement extends PanacheEntity {

	/**
	 * Le produit concerné par ce mouvement.
	 * @ManyToOne : plusieurs mouvements peuvent référencer le même produit.
	 * optional = false  ->  un mouvement DOIT avoir un produit (colonne NOT NULL).
	 */
	@ManyToOne(optional = false)
	public Article  article;

	@Enumerated(EnumType.STRING)
	@Column(nullable = false)
	public TypeMouvement type;     // ENTREE, SORTIE, AJUSTEMENT

	@Column(nullable = false)
	public int quantite;           // toujours positive

	@Column(nullable = false)

	public LocalDateTime date;

	public String motif;           // "réception", "casse", "inventaire"...

	@PrePersist
	public void avantInsertion() {
		if (date == null) {
			date = LocalDateTime.now();
		}
	}
}

File: mobile/lib/models/mouvement.dart
class Mouvement {
  final int id;

  final String type;

  final int quantite;

  final DateTime date;

  final String motif;

  Mouvement(
      {required this.id,
      required this.type,
      required this.quantite,
      required this.date,
      required this.motif});

  factory Mouvement.fromJson(json) => Mouvement(
        id: json['id'],
        type: json['type'] ?? '',
        quantite: json['quantite'] ?? 0,
        date: DateTime.parse(json['date']),
        motif: json['motif'] ?? '',
      );
}
